src/server/webapi/objects.cpp

```diff
@@ -224,6 +224,149 @@ int H_ObjectDetails(Context *context)
    return 200;
 }
 
+/**
+ * Check if an object has any descendants matching the class filter
+ */
+bool HasDescendantsMatchingClassFilter(uint32_t objectId, uint32_t userId, const std::unordered_set<int> &classFilter, std::unordered_set<uint32_t> &visited, int maxDepth, int currentDepth)
+{
+   // Prevent infinite recursion and respect maximum depth
+   if (visited.count(objectId) > 0 || currentDepth >= maxDepth)
+      return false;
+
+   visited.insert(objectId);
+
+   unique_ptr<SharedObjectArray<NetObj>> children = g_idxObjectById.getObjects(
```

Comment:
no need to scan all objects to find direct parents - they are accessible directly from object itself